aip/general/0250.md

```diff
@@ -0,0 +1,107 @@
+---
+id: 250
+state: draft
+created: 2020-09-17
+placement:
+  category: fields
+---
+
+# Short names
+
+System-generated identifiers have many benefits, but are not easy for humans
+use.
+
+## Guidance
+
+Resources with system-generated identifiers may choose to support a field
+`short_name` of type string, which **must** refer to a terse, human-friendly
+value which uniquely identifies a resource of a specific type within a
+namespace.
+
+* `short_name` **must** have clear documentation for its namespace.
+
+* `short_name` **must** support at least the character set of
+  `/^[a-z][a-z0-9.-]+[a-z0-9]$/` (lowercase
+  [RFC-1034](https://tools.ietf.org/html/rfc1034)).
+
+* `short_name` **must** have a maximum length of no less than 30 characters.
```

Comment:
We use 63 for resource IDs. What is the rationale for 30 here?

Reply:
I'm not entirely sure having this line is necessary. I added it to try and set some expectations about what the size should be (note that this is setting a minimum bound on the maximum, which isn't much of a boundary at all).

I know some companies have guidance to have an identifier be 63 characters long. This statement is compatible with that. I put 30 characters here because that is the current maximum length of a GCP project_id, which by the definition in this AIP, could be a short_name.


https://cloud.google.com/resource-manager/reference/rest/v1/projects